### PR TITLE
Add "A first look at the JIT"

### PR DESCRIPTION
**Pull request by jhogberg**
This is a work in progress like the others, and I'm posting for early feedback.

### CODE REVIEW

File: _posts/2020-11-03-a-first-look-at-the-jit.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+   tweaks depending on its arguments.
+
+   This may seem daunting at first but it's actually not that bad once you get
+   used to it. While it certainly takes a lot of code to achieve even the
+   smallest of things, it's inherently simple and easy to follow as long as
+   the code is kept short.
+
+   The downside is that every instruction needs to be implemented for each
+   architecture, but luckily there's not a lot of popular ones and we plan to
+   support the two most common ones by the time we release OTP 24; `x64` and
```
Comment (bjorng) [inline _posts/2020-11-03-a-first-look-at-the-jit.md]:
```suggestion
   support the two most common ones by the time we release OTP 24: `x64` and
```

**Review by bjorng — COMMENTED** [review]
I like it!

File: _posts/2020-11-03-a-first-look-at-the-jit.md (hunk 1 above)
Comment (jhogberg) [reply _posts/2020-11-03-a-first-look-at-the-jit.md]: I could've sworn I fixed that before posting, thanks for pointing it out!

File: _posts/2020-11-03-a-first-look-at-the-jit.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+   tweaks depending on its arguments.
+
+   This may seem daunting at first but it's actually not that bad once you get
+   used to it. While it certainly takes a lot of code to achieve even the
+   smallest of things, it's inherently simple and easy to follow as long as
+   the code is kept short.
+
+   The downside is that every instruction needs to be implemented for each
+   architecture, but luckily there's not a lot of popular ones and we plan to
+   support the two most common ones by the time we release OTP 24: `x64` and
```
Comment (KennethL) [inline _posts/2020-11-03-a-first-look-at-the-jit.md]: Are we not calling them x86_64 and ARM64? I don't think we are aiming for 32 bit architectures. Should we dare to promise ARM64 to OTP 24? , Maybe? But should we flag for this as a promise here, we can maybe give a hint.

File: _posts/2020-11-03-a-first-look-at-the-jit.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+     * if the zero flag is clear (the list is empty). */
+    a.jnz(labels[Fail.getValue()]);
+
+    /* Unlike the interpreter there's no need to jump to
+     * the next instruction on success as it immediately
+     * follows this one. */
+}
+```
+
+The code generated by this template looks almost identical, let's say our
```
Comment (KennethL) [inline _posts/2020-11-03-a-first-look-at-the-jit.md]: looks identical to what?

**Comment by KennethL** [comment]
Looks good, some small comments inline.

File: _posts/2020-11-03-a-first-look-at-the-jit.md (hunk 2 above)
Comment (KennethL) [inline _posts/2020-11-03-a-first-look-at-the-jit.md]: x86_64
The plus side is that even if we have to implement every instruction manually for each architecture there are only 2 ones that are really popular and the interpreter will still be available for all other architectures.

**Review by ferd — APPROVED** [review]

**Review by garazdawi — APPROVED** [review]

**Review by KennethL — COMMENTED** [review]
Looks good just a few comments inline

**Pull request merged**